cmd/cluster-agent/subcommands/start/command.go

```diff
@@ -168,8 +169,14 @@ func start(log log.Component, config config.Component, telemetry telemetry.Compo
 		}
 	}
 
+	// Setup the leader forwarder for language detection and cluster checks
+	var leaderForwarder *apidca.LeaderForwarder
```

Comment:
Here the leader engine still does not exist but I think we might not need it

Reply:
This choice was made because both clc runners and pld handler need the leader forwarder. However, we don't want to duplicate connections.